#!/usr/bin/env python
PACKAGE = "strategy"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

## Game Start or not
gen.add("game_start",   bool_t,   0, "Game Start or not",  False)

## Select Team between Cyan or Magenta
side_enum = gen.enum([ gen.const("Cyan_Team",    str_t, "Cyan",    "Attack Yellow Goal"),
                       gen.const("Magenta_Team", str_t, "Magenta", "Attack Blue Goal")],
                     "An enum to set Team")
gen.add("side", str_t, 0, "A side parameter which is edited via an enum", "Cyan", edit_method=side_enum)

## Select Game State
game_state_enum = gen.enum([ gen.const("Kick_Off",     str_t, "Kick_Off",     "Kick_Off"),
                             gen.const("Free_Kick",    str_t, "Free_Kick",    "Free_Kick"),
                             gen.const("Free_Ball",    str_t, "Free_Ball",    "Free_Ball"),
                             gen.const("Throw_In",     str_t, "Throw_In",     "Throw_In"),
                             gen.const("Coner_Kick",   str_t, "Coner_Kick",   "Coner_Kick"),
                             gen.const("Penalty_Kick", str_t, "Penalty_Kick", "Penalty_Kick")],
                             "An enum to set Game State")
gen.add("game_state", str_t, 0, "Select Game State before game started", "Kick_Off", edit_method=game_state_enum)

## Robot Speed
gen.add("minimum_w", double_t,   0, "minimum speed of angular velocity",  0.2, 0, 5)
gen.add("maximum_w", double_t,   0, "Maximum speed of angular velocity",  100, 10, 100)
# gen.add("kp_w", double_t,   0, "Kp value of angular velocity",  0.4, 0, 1)
# gen.add("ki_w", double_t,   0, "Ki value of angular velocity",  0, 0, 1)
# gen.add("kd_w", double_t,   0, "Kd value of angular velocity",  0, 0, 1)

gen.add("minimum_v", double_t,   0, "minimum speed of velocity",  0.2, 0, 5)
gen.add("maximum_v", double_t,   0, "Maximum speed of velocity",  100, 10, 100)
# gen.add("kp_v", double_t,   0, "Kp value of velocity",  0.4, 0, 1)
# gen.add("ki_v", double_t,   0, "Ki value of velocity",  0, 0, 1)
# gen.add("kd_v", double_t,   0, "Kd value of velocity",  0, 0, 1)

## Strategy Configs
gen.add("ballhandle_dis", int_t,   0, "Condition of Handleing ball", 30, 0, 50)
gen.add("ballhandle_ang", int_t,   0, "Condition of Handleing ball", 10, 0, 50)

exit(gen.generate(PACKAGE, "strategy", "GameState"))
